migrator/migrations/m_181_to_m_182_remove_default_scope_manager_role/migration_impl.go

```diff
@@ -0,0 +1,327 @@
+package m181tom182
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	apiTokenStore "github.com/stackrox/rox/migrator/migrations/m_181_to_m_182_remove_default_scope_manager_role/apitokenstore"
+	groupStore "github.com/stackrox/rox/migrator/migrations/m_181_to_m_182_remove_default_scope_manager_role/groupstore"
+	permissionSetStore "github.com/stackrox/rox/migrator/migrations/m_181_to_m_182_remove_default_scope_manager_role/permissionsetstore"
+	roleStore "github.com/stackrox/rox/migrator/migrations/m_181_to_m_182_remove_default_scope_manager_role/rolestore"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sac"
+)
+
+// Pre- and Post- migration role and permission set names
+const (
+	ScopeManagerRoleName = "Scope Manager"
+
+	deprecatedPrefix            = "[DEPRECATED] "
+	deprecatedDescriptionSuffix = ". DEPRECATED, please use \"Vulnerability Report Creator\" " +
+		"instead for vulnerability report management purposes"
+
+	scopeManagerPermissionSetID = "ffffffff-ffff-fff4-f5ff-fffffffffffb"
+)
+
+const (
+	batchSize = 500
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	errNotFound = errors.New("not found")
+)
+
+func migrate(database *types.Databases) error {
+	ctx := sac.WithAllAccess(context.Background())
+
+	apiTokenStorage := apiTokenStore.New(database.PostgresDB)
+	groupStorage := groupStore.New(database.PostgresDB)
+	permissionSetStorage := permissionSetStore.New(database.PostgresDB)
+	roleStorage := roleStore.New(database.PostgresDB)
+
+	var err *multierror.Error
+	canRemoveRole := true
+	canRemovePermissionSet := true
+	// Check whether the old default role is used
+	usesScopeManagerRole, useRoleCheckErr := isScopeManagerRoleReferenced(ctx, apiTokenStorage, groupStorage)
+	if useRoleCheckErr != nil {
+		canRemoveRole = false
+		err = multierror.Append(err, useRoleCheckErr)
+	}
+	if usesScopeManagerRole {
+		addRoleErr := addDeprecatedScopeManagerRole(ctx, roleStorage)
+		if addRoleErr != nil {
+			canRemoveRole = false
+			err = multierror.Append(err, addRoleErr)
```

Comment:
With the changes above to fail fast:
```suggestion
		if addRoleErr := addDeprecatedScopeManagerRole(ctx, roleStorage); addRoleErr != nil {
			return err
		}
```